byterun/caml/osdeps.h

```diff
@@ -79,18 +79,54 @@ extern char * caml_dlerror(void);
 /* Add to [contents] the (short) names of the files contained in
    the directory named [dirname].  No entries are added for [.] and [..].
    Return 0 on success, -1 on error; set errno in the case of error. */
-extern int caml_read_directory(char * dirname, struct ext_table * contents);
+extern int caml_read_directory(charnat * dirname, struct ext_table * contents);
 
 /* Recover executable name if possible (/proc/sef/exe under Linux,
    GetModuleFileName under Windows).  Return NULL on error,
    string allocated with [caml_stat_alloc] on success. */
-extern char * caml_executable_name(void);
+extern charnat * caml_executable_name(void);
 
 /* Secure version of [getenv]: returns NULL if the process has special
    privileges (setuid bit, setgid bit, capabilities).
 */
-extern char *caml_secure_getenv(char const *var);
+extern charnat *caml_secure_getenv(charnat const *var);
 
 #endif /* CAML_INTERNALS */
 
+/* Windows Unicode support */
+
+#ifdef _WIN32
```

Comment:
Should these be inside the `CAML_INTERNALS` block (i.e. are expecting to expose these to C stubs at this stage?)

Reply:
Yes, fixed it.

Reply:
Actually, we need these functions from `otherlibs`, so they can't be protected by `CAML_INTERNALS`.